taskfiles/deps.yaml

```
@@ -0,0 +1,77 @@
+version: "3"
+
+includes:
+  utils: "../tools/yscope-dev-utils/exports/taskfiles/utils/utils.yaml"
+
+vars:
+  G_DEPS_DIR: "{{.G_BUILD_DIR}}/deps"
+
+  # This path must be kept in-sync with its usage in CMakeLists.txt and examples/CMakeLists.txt.
+  G_DEPS_CMAKE_SETTINGS_DIR: "{{.G_DEPS_DIR}}/cmake-settings"
+
+tasks:
+  default:
+    deps:
+      - "all"
+
+  all:
+    run: "once"
+    cmds:
+      - task: "utils:cmake:install-deps-and-generate-settings"
+        vars:
+          CMAKE_SETTINGS_DIR: "{{.G_DEPS_CMAKE_SETTINGS_DIR}}"
+          DEP_TASK: "all-parallel"
+
+  all-parallel:
+    internal: true
+    run: "once"
+    deps:
+      - "Catch2"
+      - "fmt"
+      - "Microsoft.GSL"
+
+  Catch2:
+    internal: true
```

```suggestion
    internal: true
    vars:
      LIB_NAME: "Catch2"
```

Ditto for the other libs.